packages/bento-design-system/src/TextField/createTextField.tsx

```diff
@@ -31,34 +45,69 @@ export function createTextField(config: InputConfig, { Field }: { Field: FieldTy
       inputRef
     );
 
+    const [showPassword, setShowPassword] = useState(false);
+    const passwordIcon = showPassword
+      ? config.passwordHideIcon ?? IconEyeClosed
+      : config.passwordShowIcon ?? IconEye;
+    const passwordIconLabel = showPassword
+      ? props.hidePasswordLabel ?? defaultMessages.TextField.hidePasswordLabel
+      : props.showPasswordLabel ?? defaultMessages.TextField.showPasswordLabel;
+
+    const type = props.type === "password" && !showPassword ? "password" : "text";
+
     return (
       <Field
         {...props}
         labelProps={labelProps}
         assistiveTextProps={descriptionProps}
         errorMessageProps={errorMessageProps}
       >
-        <Box
-          as="input"
-          type={props.type || "text"}
-          ref={inputRef}
-          {...inputProps}
-          // NOTE(gabro): this is to please TS, since the inputProps type is very broad
-          color={undefined}
-          width={undefined}
-          height={undefined}
-          borderRadius={config.radius}
-          paddingX={config.paddingX}
-          paddingY={config.paddingY}
-          className={[
-            inputRecipe({ validation: validationState || "notSet" }),
-            bodyRecipe({
-              color: props.disabled ? "disabled" : "default",
-              weight: "default",
-              size: config.fontSize,
-            }),
-          ]}
-        />
+        <Box position="relative" display="flex">
+          <Box
+            as="input"
+            ref={inputRef}
+            {...inputProps}
+            type={type}
+            // NOTE(gabro): this is to please TS, since the inputProps type is very broad
+            color={undefined}
+            width={undefined}
+            height={undefined}
+            borderRadius={config.radius}
+            paddingX={config.paddingX}
+            paddingY={config.paddingY}
+            className={[
+              inputRecipe({ validation: validationState || "notSet" }),
+              bodyRecipe({
+                color: props.disabled ? "disabled" : "default",
+                weight: "default",
+                size: config.fontSize,
+              }),
+            ]}
+            style={{ paddingRight: rightAccessoryWidth, flexGrow: 1 }}
+          />
+          {props.type === "password" && (
+            <Box
+              ref={rightAccessoryRef}
+              position="absolute"
+              display="flex"
+              justifyContent="center"
+              alignItems="center"
+              paddingX={16}
+              top={0}
+              bottom={0}
+              right={0}
+            >
+              <IconButton
+                size={config.passwordIconSize ?? 24}
```

Comment:
The default should not live here, but in the `defaultConfig`

Reply:
Also `passwordHideIcon` and `passwordShowIcon`?

Reply:
yes, actually, good point